omigami/data_models.py

```diff
@@ -0,0 +1,109 @@
+# TODO: rename as data_models
+
+from typing import Iterable, List, Dict, Union
+from dataclasses import dataclass
+from omigami.data_types import NumpyArray
+from omigami.model import Estimator
+
+
+@dataclass
+class Split:
+    id: int
+    train_indices: NumpyArray
+    test_indices: NumpyArray
+
+
+@dataclass
+class InputData:
+    X: NumpyArray
+    y: NumpyArray
+    groups: NumpyArray
+    n_features: int = None
+
+    def __post_init__(self):
+        if self.n_features is None:
+            self.n_features = self.X.shape[1]
```

Comment:
Hm didn't know Python could do this. But wouldn't this be nicer:
```
@property
def n_features():
    return self.X.shape[1]
```

Reply:
This is a dataclass thing I believe, not for every python class. I will see how it looks like with your suggestion